packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -167,6 +183,14 @@ export default createRule<Options, MessageId>({
         if (invalidAncestor.type === AST_NODE_TYPES.ReturnStatement) {
           // handle return statement
 
+          if (options.ignoreVoidInVoid) {
+            if (
+              targetNodeFunctionAncestorNodeHasVoidReturnType(invalidAncestor)
+            ) {
+              return;
+            }
+          }
```

Comment:
[Style] Same as here: https://github.com/typescript-eslint/typescript-eslint/pull/8632#discussion_r1518572544

```suggestion
          if (options.ignoreVoidInVoid && targetNodeFunctionAncestorNodeHasVoidReturnType(invalidAncestor)) {
            return;
          }
```

WDYT about inlining this if statement?